src/m365/commands/docs.spec.ts

```diff
@@ -0,0 +1,82 @@
+import Command from '../../Command';
+import commands from './commands';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { Logger } from '../../cli/Logger';
+import * as open from 'open';
+import { telemetry } from '../../telemetry';
+import { pid } from '../../utils/pid';
+import { session } from '../../utils/session';
+import { sinonUtil } from '../../utils/sinonUtil';
+import { Cli } from '../../cli/Cli';
+const packageJSON = require('../../../package.json');
+const command: Command = require('./docs');
+
+describe(commands.DOCS, () => {
+  let log: any[];
+  let logger: Logger;
+  let cli: Cli;
+  let loggerLogSpy: sinon.SinonSpy;
+  let getSettingWithDefaultValueStub: sinon.SinonStub;
+  let openStub: sinon.SinonStub;
+
+  before(() => {
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+  });
+
+  beforeEach(() => {
+    log = [];
+    cli = Cli.getInstance();
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any)._open = open;
+    openStub = sinon.stub(command as any, '_open').callsFake(() => Promise.resolve(null));
+    getSettingWithDefaultValueStub = sinon.stub(cli, 'getSettingWithDefaultValue').returns(false);
+  });
+
+  afterEach(() => {
+    loggerLogSpy.restore();
+    getSettingWithDefaultValueStub.restore();
+    openStub.restore();
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
```

Comment:
A new approach of restoring all stubs is `sinon.restore()`. Let's try that.

Reply:
Using `sinon.restore` in `afterEach` hook seems to be working. I will remove individual restores and also `sinonUtil.restore` now.

Reply:
You can use specific restores in the `afterEach` hook, just make sure that the `each` hook uses `sinon.restore()`.